### PR TITLE
Drop metadata_encoding_version.

### PR DESCRIPTION
**Pull request by cjgillot**
Part of #85153

r? @Aaron1011

### CODE REVIEW

File: compiler/rustc_metadata/src/rmeta/mod.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ const METADATA_VERSION: u8 = 5;
 /// This header is followed by the position of the `CrateRoot`,
 /// which is encoded as a 32-bit big-endian unsigned integer,
 /// and further followed by the rustc version string.
-crate const METADATA_HEADER: &[u8; 8] = &[b'r', b'u', b's', b't', 0, 0, 0, METADATA_VERSION];
+pub const METADATA_HEADER: &[u8; 8] = &[b'r', b'u', b's', b't', 0, 0, 0, METADATA_VERSION];
```
Comment (bjorn3) [inline compiler/rustc_metadata/src/rmeta/mod.rs:54]: Just a wild guess, but maybe this fixes the perf regression?
```suggestion
pub const METADATA_HEADER: &[u8] = &[b'r', b'u', b's', b't', 0, 0, 0, METADATA_VERSION];
```

**Comment by ehuss** [comment]
@bors retry
aarch64-gnu self-hosted runner mysteriously died after 25 minutes, no logs available.

**Pull request merged**